plugins/outputs/microsoft_fabric/microsoft_fabric.go

```diff
@@ -0,0 +1,126 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package microsoft_fabric
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type fabricOutput interface {
+	Init() error
+	Connect() error
+	Write(metrics []telegraf.Metric) error
+	Close() error
+}
+
+type MicrosoftFabric struct {
+	ConnectionString string          `toml:"connection_string"`
+	Log              telegraf.Logger `toml:"-"`
+	Timeout          config.Duration `toml:"timeout"`
+
+	eventhouse   *eventhouse
+	eventstream  *eventstream
+	activePlugin fabricOutput
+}
+
+func (*MicrosoftFabric) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *MicrosoftFabric) Init() error {
+	connectionString := m.ConnectionString
+
+	if connectionString == "" {
+		return errors.New("endpoint must not be empty")
+	}
```

Comment:
There is no need to use a local alias!

```suggestion
	if m.Connection == "" {
		return errors.New("'connection' must not be empty")
	}
```